### PR TITLE
Ports: Update dosbox-staging to 0.77.1 + add libopus 1.3.1 and opusfile 0.12

### PR DESCRIPTION
**Pull request by kennethmyhra**
This version of `dosbox-staging` uses the meson build system.

Previous versions of `dosbox-staging` started in windowed mode with a resolution of 320x200. This version starts in windowed mode with a resolution of 640x480.

Audio was stuttering a bit in previous versions, but it sounds like it might have become a bit worse. This is probably because of higher CPU usage and the audio server/client not able to keep up.

Added a `post_install()` section to package script which outputs a link to `dosbox-staging`'s release notes.

This also adds two new ports: `opusfile`, and `libopus`.

We are not yet at the latest version, but this should probably make it bit easier to upgrade. For this PR I had to draw a line at moving `dosbox-staging` to the meson build system and adding two new required ports.

### CODE REVIEW

File: Ports/libopus/package.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='libopus'
+version='1.3.1'
+workdir='opus-1.3.1'
+useconfigure='true'
+files=" https://archive.mozilla.org/pub/opus/opus-${version}.tar.gz opus-${version}.tar.gz 65b58e1e25b2a114157014736a3d9dfeaad8d41be1c8179866f144a2fb44ff9d"
```
Comment (GMTA) [inline Ports/libopus/package.sh:6]: Stray space at the beginning of this string

File: Ports/libopus/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/config.sub b/config.sub
+index 932128b337eeec7cdc60fd8e0f361531e80295f5..1da4fcc025afdbbf675c621f73aebba27f5c27e1 100755
+--- a/config.sub
++++ b/config.sub
+@@ -1417,7 +1417,7 @@ case $os in
+ 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
+ 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
+ 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
++	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -serenity*)
```
Comment (GMTA) [inline Ports/libopus/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch:19]: Just to be sure: `use_fresh_config_sub` did not work here?

Comment (kennethmyhra) [reply Ports/libopus/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch]: `use_fresh_config_sub` does indeed work

File: Ports/opusfile/package.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='opusfile'
+version='0.12'
+useconfigure='true'
+files="https://downloads.xiph.org/releases/opus/opusfile-${version}.tar.gz opusfile-${version}.tar.gz 118d8601c12dd6a44f52423e68ca9083cc9f2bfe72da7a8c1acb22a80ae3550b"
+auth_type='sha256'
+depends=(
+  'libogg'
+  'libopus'
```
Comment (GMTA) [inline Ports/opusfile/package.sh:9]: These should be 4-space indents I believe

File: Ports/opusfile/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/config.sub b/config.sub
+index 1d8e98bcee23a0421e4fafe9a6c9ac75180cff25..e1d397a3c3126a96936406663df0a363846a415e 100755
+--- a/config.sub
++++ b/config.sub
+@@ -1390,7 +1390,7 @@ case $os in
+ 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
+ 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+ 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
+-	      | -midnightbsd*)
++	      | -midnightbsd* | -serenity*)
```
Comment (GMTA) [inline Ports/opusfile/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch:19]: `use_fresh_config_sub`?

Comment (kennethmyhra) [reply Ports/opusfile/patches/0001-Recognize-SerenityOS-as-a-host-platform.patch]: Absolutely. I need to be more aware of this option. Thanks!

**Comment by kennethmyhra** [comment]
Managed to mess up the rebase for a moment there. Hopefully all good now, and all things addressed

**Review by GMTA — APPROVED** [review]

**Pull request merged**